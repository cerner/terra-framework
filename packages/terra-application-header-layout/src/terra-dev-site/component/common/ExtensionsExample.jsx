import React from 'react';
import PropTypes from 'prop-types';
import PlaceHolder from './Placeholder';

const propTypes = {
  size: PropTypes.string,
};

const ExtensionsExample = ({ size }) => {
  let text = 'Extensions';
  let width = '150px';
  if (size === 'tiny') {
    text = 'E';
    width = '50px';
  } else if (size === 'small') {
    text = 'Ext';
    width = '100px';
  }
  return (
    <PlaceHolder text={text} width={width} />
  );
};

ExtensionsExample.propTypes = propTypes;

export default ExtensionsExample;
