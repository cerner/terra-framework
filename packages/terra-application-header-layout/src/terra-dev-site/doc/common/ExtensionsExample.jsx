import React from 'react';
import PropTypes from 'prop-types';
import PlaceHolder from './Placeholder';

const propTypes = {
  size: PropTypes.string,
};

const ExtensionsExample = ({ size }) => {
  let text = 'Extensions';
  if (size === 'tiny') {
    text = 'E';
  } else if (size === 'small') {
    text = 'Ext';
  }
  return (
    <PlaceHolder text={text} size={size} />
  );
};

ExtensionsExample.propTypes = propTypes;

export default ExtensionsExample;
