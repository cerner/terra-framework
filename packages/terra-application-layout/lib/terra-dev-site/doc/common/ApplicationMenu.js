"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _RoutingStackDelegate = _interopRequireDefault(require("terra-navigation-layout/lib/RoutingStackDelegate"));
var _terraButton = _interopRequireDefault(require("terra-button"));
var _terraContentContainer = _interopRequireDefault(require("terra-content-container"));
var _bind = _interopRequireDefault(require("classnames/bind"));
var _ApplicationMenuModule = _interopRequireDefault(require("./ApplicationMenu.module.scss"));
var _RoutingMenu = _interopRequireDefault(require("../../../menu/RoutingMenu"));
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
var cx = _bind.default.bind(_ApplicationMenuModule.default);
var ApplicationMenu = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(ApplicationMenu, _React$Component);
  var _super = _createSuper(ApplicationMenu);
  function ApplicationMenu(props) {
    var _this;
    (0, _classCallCheck2.default)(this, ApplicationMenu);
    _this = _super.call(this, props);
    var items = [];
    for (var i = 0; i < 20; i += 1) {
      items.push({
        text: "Item ".concat(i),
        path: "".concat(props.baseUrl, "/item_").concat(i)
      });
    }
    if (props.includeNestedMenu) {
      items.push({
        text: 'Nested Menu',
        path: "".concat(props.baseUrl, "/nested"),
        hasSubMenu: true
      });
    }
    var menuItems = items.map(function (item) {
      return {
        key: item.path,
        text: item.text,
        path: item.path,
        hasSubMenu: item.hasSubMenu
      };
    });
    _this.state = {
      menuItems: menuItems
    };
    return _this;
  }
  (0, _createClass2.default)(ApplicationMenu, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
        menuName = _this$props.menuName,
        layoutConfig = _this$props.layoutConfig,
        routingStackDelegate = _this$props.routingStackDelegate;
      var menuItems = this.state.menuItems;
      return /*#__PURE__*/_react.default.createElement(_terraContentContainer.default, {
        footer: /*#__PURE__*/_react.default.createElement("div", {
          className: cx('footer')
        }, /*#__PURE__*/_react.default.createElement(_terraButton.default, {
          text: "Custom Event 1",
          isBlock: true,
          onClick: function onClick() {
            var evt = document.createEvent('CustomEvent');
            evt.initCustomEvent('applicationMenu.itemSelected', false, false, 'Custom Event 1');
            document.dispatchEvent(evt);
            if (layoutConfig && layoutConfig.toggleMenu) {
              layoutConfig.toggleMenu();
            }
          },
          className: cx('custom-event-button')
        }), /*#__PURE__*/_react.default.createElement(_terraButton.default, {
          text: "Custom Event 2",
          isBlock: true,
          onClick: function onClick() {
            var evt = document.createEvent('CustomEvent');
            evt.initCustomEvent('applicationMenu.itemSelected', false, false, 'Custom Event 2');
            document.dispatchEvent(evt);
            if (layoutConfig && layoutConfig.toggleMenu) {
              layoutConfig.toggleMenu();
            }
          }
        })),
        fill: true
      }, /*#__PURE__*/_react.default.createElement(_RoutingMenu.default, {
        title: menuName,
        menuItems: menuItems,
        routingStackDelegate: routingStackDelegate,
        layoutConfig: layoutConfig
      }));
    }
  }]);
  return ApplicationMenu;
}(_react.default.Component);
ApplicationMenu.propTypes = {
  layoutConfig: _propTypes.default.shape({
    toggleMenu: _propTypes.default.func,
    togglePin: _propTypes.default.func,
    menuIsPinned: _propTypes.default.bool
  }),
  routingStackDelegate: _RoutingStackDelegate.default.propType,
  includeNestedMenu: _propTypes.default.bool,
  menuName: _propTypes.default.string,
  baseUrl: _propTypes.default.string
};
var _default = ApplicationMenu;
exports.default = _default;