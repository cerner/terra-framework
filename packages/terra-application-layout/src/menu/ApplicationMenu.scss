:local {
  .application-menu {
    background-color: var(--terra-application-menu-background-color, transparent);
    height: 100%;
    position: relative;
  }

  .menu-header {
    background-color: var(--terra-application-menu-header-background-color, #1b7fcd);
    position: relative;
    width: 100%;
  }
}
