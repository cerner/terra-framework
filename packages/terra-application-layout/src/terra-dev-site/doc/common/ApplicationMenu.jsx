import React from 'react';
import PropTypes from 'prop-types';
import RoutingStackDelegate from 'terra-navigation-layout/lib/RoutingStackDelegate';
import Button from 'terra-button';
import ContentContainer from 'terra-content-container';
import classNames from 'classnames/bind';
import styles from './ApplicationMenu.scss';

import RoutingMenu from '../../../menu/RoutingMenu';

const cx = classNames.bind(styles);

class ApplicationMenu extends React.Component {
  constructor(props) {
    super(props);

    const items = [];

    for (let i = 0; i < 20; i += 1) {
      items.push({
        text: `Item ${i}`,
        path: `${props.baseUrl}/item_${i}`,
      });
    }

    if (props.includeNestedMenu) {
      items.push({
        text: 'Nested Menu',
        path: `${props.baseUrl}/nested`,
        hasSubMenu: true,
      });
    }

    const menuItems = items.map(item => ({
      key: item.path,
      text: item.text,
      path: item.path,
      hasSubMenu: item.hasSubMenu,
    }));

    this.state = {
      menuItems,
    };
  }

  render() {
    const { menuName, layoutConfig, routingStackDelegate } = this.props;
    const { menuItems } = this.state;

    return (
      <ContentContainer
        footer={(
          <div className={cx('footer')}>
            <Button
              text="Custom Event 1"
              isBlock
              onClick={() => {
                const evt = document.createEvent('CustomEvent');
                evt.initCustomEvent('applicationMenu.itemSelected', false, false, 'Custom Event 1');
                document.dispatchEvent(evt);

                if (layoutConfig && layoutConfig.toggleMenu) {
                  layoutConfig.toggleMenu();
                }
              }}
              className={cx('custom-event-button')}
            />
            <Button
              text="Custom Event 2"
              isBlock
              onClick={() => {
                const evt = document.createEvent('CustomEvent');
                evt.initCustomEvent('applicationMenu.itemSelected', false, false, 'Custom Event 2');
                document.dispatchEvent(evt);

                if (layoutConfig && layoutConfig.toggleMenu) {
                  layoutConfig.toggleMenu();
                }
              }}
            />
          </div>
        )}
        fill
      >
        <RoutingMenu
          title={menuName}
          menuItems={menuItems}
          routingStackDelegate={routingStackDelegate}
          layoutConfig={layoutConfig}
        />
      </ContentContainer>
    );
  }
}

ApplicationMenu.propTypes = {
  layoutConfig: PropTypes.shape({
    toggleMenu: PropTypes.func,
    togglePin: PropTypes.func,
    menuIsPinned: PropTypes.bool,
  }),
  routingStackDelegate: RoutingStackDelegate.propType,
  includeNestedMenu: PropTypes.bool,
  menuName: PropTypes.string,
  baseUrl: PropTypes.string,
};

export default ApplicationMenu;
