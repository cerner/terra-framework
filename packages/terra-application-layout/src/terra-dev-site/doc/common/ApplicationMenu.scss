@import './ApplicationLayoutDocCommon';

:local {
  .footer {
    @extend %common-container-content;
    border-top: 1px solid #d3d3d3;
  }

  .custom-event-button {
    margin-bottom: 5px;
  }
}
