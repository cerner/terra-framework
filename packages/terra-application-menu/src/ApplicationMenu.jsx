import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import AppDelegate from 'terra-app-delegate';
import ApplicationMenuLayout from 'terra-application-menu-layout';
// import { ApplicationMenuUtility } from 'terra-application-utility';
import { ApplicationMenuName } from 'terra-application-name';
import RoutingStackDelegate from 'terra-navigation-layout/lib/RoutingStackDelegate';

import 'terra-base/lib/baseStyles';

import ApplicationMenuUtility from './mock-components/MockApplicationUtility';

import styles from './ApplicationMenu.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The AppDelegate instance that will be propagated to the components presented within the NavigationLayout.
   */
  app: AppDelegate.propType,
  /**
   * The element to be placed within the fill flex styled content area.
   */
  content: PropTypes.element,
  /**
   * Layout config provided from the Layout component.
   */
  layoutConfig: PropTypes.shape({
    size: PropTypes.string,
    toggleMenu: PropTypes.func,
    menuIsOpen: PropTypes.bool,
    togglePin: PropTypes.bool,
    menuIsPinned: PropTypes.bool,
  }).isRequired,
  /**
   * Configutation values for the ApplicationName component.
   */
  nameConfig: PropTypes.shape({
    accessory: PropTypes.element,
    title: PropTypes.string,
  }),
  /**
   * Delegate prop that is provided by the NavigationLayout.
   */
  routingStackDelegate: RoutingStackDelegate.propType,
  /**
   * Configration to be passed to the ApplicationUtility component.
   */
  utilityConfig: PropTypes.shape({
    userName: PropTypes.string,
    userPhoto: PropTypes.element,
    userDetails: PropTypes.string,
    onUtilityChange: PropTypes.func,
  }).isRequired,
};

const defaultProps = {
  nameConfig: {},
};

class ApplicationMenu extends React.Component {
  constructor(props) {
    super(props);
    this.onDiscloseUtilty = this.onDiscloseUtilty.bind(this);
  }

  onDiscloseUtilty(utility) {
    if (this.props.app && utility) {
      this.props.app.disclose({
        preferredType: 'modal',
        size: 'small',
        content: {
          component: utility,
          key: 'application-utility-menu',
        },
      });
    }
  }

  render() {
    const {
      app,
      content,
      layoutConfig,
      nameConfig,
      routingStackDelegate,
      utilityConfig,
      ...customProps
    } = this.props;

    const menuClassNames = cx([
      'application-menu',
      customProps.className,
    ]);

    let header;
    if (nameConfig.accessory || nameConfig.title) {
      header = <ApplicationMenuName accessory={nameConfig.accessory} title={nameConfig.title} />;
    }

    const footer = <ApplicationMenuUtility {...utilityConfig} onDiscloseUtilityMenu={this.onDiscloseUtilty} />;

    let clonedContent;
    if (content) {
      clonedContent = React.cloneElement(content, { layoutConfig });
    }

    return (
      <div {...customProps} className={menuClassNames}>
        <ApplicationMenuLayout
          header={header}
          content={clonedContent}
          footer={footer}
        />
      </div>
    );
  }
}

ApplicationMenu.propTypes = propTypes;
ApplicationMenu.defaultProps = defaultProps;

export default ApplicationMenu;
