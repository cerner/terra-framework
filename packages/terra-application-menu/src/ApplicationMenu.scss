:local {
  .application-menu {
    background-color: var(--application-menu-background-color, transparent);
    height: 100%;
    position: relative;
    width: 100%;
  }
}
