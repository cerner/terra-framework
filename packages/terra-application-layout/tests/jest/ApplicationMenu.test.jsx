import React from 'react';

import ThemeContextProvider from 'terra-theme-context/lib/ThemeContextProvider';
import ApplicationMenu from '../../src/menu/_ApplicationMenu';
import ExtensionsExample from '../../src/terra-dev-site/test/common/ExtensionsExample';
import ContentExample from '../../src/terra-dev-site/test/common/ContentExample';

describe('ApplicationHeader', () => {
  it('should render default component', () => {
    const testHeader = (
      <ApplicationMenu
        routingStackDelegate={{}}
        layoutConfig={{ size: 'small' }}
      />
    );

    const result = shallow(testHeader);

    expect(result).toMatchSnapshot();
  });

  it('should render props', () => {
    const testHeader = (
      <ApplicationMenu
        id="test-menu"
        layoutConfig={{ size: 'small' }}
        nameConfig={{ title: 'app-test-title' }}
        routingStackDelegate={{}}
        utilityConfig={{
          title: 'test-util-title',
          initialSelectedKey: 'test-menu',
          onChange: () => {},
          menuItems: [
            {
              key: 'test-menu',
              contentLocation: 'body',
              title: 'test-menu-title',
            },
          ],
        }}
        extensions={<ExtensionsExample layoutConfig={{ size: 'small' }} />}
        content={<ContentExample />}
      />
    );

    const result = shallow(testHeader);

    expect(result).toMatchSnapshot();
  });

  it('correctly applies the theme context className', () => {
    const testHeader = mount(
      <ThemeContextProvider theme={{ className: 'orion-fusion-theme' }}>
        <ApplicationMenu
          routingStackDelegate={{}}
          layoutConfig={{ size: 'small' }}
        />
      </ThemeContextProvider>,
    );
    expect(testHeader).toMatchSnapshot();
  });
});
