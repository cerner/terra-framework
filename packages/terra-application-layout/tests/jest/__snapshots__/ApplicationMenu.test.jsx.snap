// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationHeader correctly applies the theme context className 1`] = `
<ThemeContextProvider
  theme={
    Object {
      "className": "orion-fusion-theme",
    }
  }
>
  <withDisclosureManager(ApplicationMenu)
    layoutConfig={
      Object {
        "size": "small",
      }
    }
    routingStackDelegate={Object {}}
  >
    <ApplicationMenu
      layoutConfig={
        Object {
          "size": "small",
        }
      }
      routingStackDelegate={Object {}}
    >
      <div
        className="application-menu orion-fusion-theme"
      >
        <ApplicationMenuLayout
          extensions={null}
          footer={null}
          header={null}
        >
          <div
            className="menu fill"
          >
            <div
              className="fill body"
            />
          </div>
        </ApplicationMenuLayout>
      </div>
    </ApplicationMenu>
  </withDisclosureManager(ApplicationMenu)>
</ThemeContextProvider>
`;

exports[`ApplicationHeader should render default component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`ApplicationHeader should render props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
